internal/message/consumer.go

```
@@ -0,0 +1,237 @@
+		*sqs.DeleteMessageOutput, error,
+	)
+}
+
+type ReceiverDeleter interface {
+	ReceiverAPI
+	DeleterAPI
+}
+
+func GetQueueUrl(c context.Context, api ReceiverAPI, input *sqs.GetQueueUrlInput) (*sqs.GetQueueUrlOutput, error) {
+	return api.GetQueueUrl(c, input)
+}
+
+func GetMessages(c context.Context, api ReceiverAPI, input *sqs.ReceiveMessageInput) (
+	*sqs.
+		ReceiveMessageOutput, error,
+) {
+	return api.ReceiveMessage(c, input)
+}
+
+func DeleteMessage(c context.Context, api DeleterAPI, input *sqs.DeleteMessageInput) (*sqs.DeleteMessageOutput, error) {
+	return api.DeleteMessage(c, input)
+}
+
+func (n *Consumer) Consume(ctx context.Context) {
+	getQueueInput := &sqs.GetQueueUrlInput{
+		QueueName: &n.queueName,
+	}
+
+	queueUrlResult, err := GetQueueUrl(ctx, n.sqsClient, getQueueInput)
+	if err != nil {
+		n.Error("got an error while getting queue url:", zap.Error(err))
+		return
+	}
+
+	queueUrl := queueUrlResult.QueueUrl
+
+	getMessageInput := &sqs.ReceiveMessageInput{
+		QueueUrl:            queueUrl,
+		MaxNumberOfMessages: 10,
+	}
+
+	msgResult, err := GetMessages(ctx, n.sqsClient, getMessageInput)
+	if err != nil {
+		n.Error("got an error while receiving messages:", zap.Error(err))
+		return
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			msgResult, err = GetMessages(ctx, n.sqsClient, getMessageInput)
+			if err != nil {
+				n.Error("got an error while receiving messages:", zap.Error(err))
+				return
+			}
+			for _, v := range msgResult.Messages {
+				err = n.ProcessMessage(ctx, v)
+				if err != nil {
+					n.Logger.Error("error processing message: ", zap.String("error_message_id", *v.MessageId))
+				} else {
+					err = n.deleteMessage(ctx, v.ReceiptHandle)
+					if err != nil {
+						n.Logger.Error(
+							"could not delete message: ", zap.String("failed_delete_message_id", *v.MessageId),
+						)
```

Update this logic so that errors are not getting swallowed silently.